### PR TITLE
Added onCreate prop for creatable inputPicker

### PR DESCRIPTION
**Pull request by BossBele**
This feature will help when trying to do something with a value not previously on the data list. Use case involves, fetching an array of data as data list and adding a created value to the database so that the user does not have to create it again.
Created from ISSUE #1567 

The onCreate prop is a function that fires when the `creatable` `inputPicker` has a new created value that is not in the list. The function passes values as the onSelect prop does:
`onCreate={(value, item, event) => null}`

### Example:
```
<InputPicker
creatable
onCreate={(value, item, event) => addToDatabase(value)} />
```

### CODE REVIEW

File: src/InputPicker/InputPicker.tsx [hunk 1]
```diff
@@ -313,10 +314,11 @@ const InputPicker: PickerComponent<InputPickerProps> = React.forwardRef(
 
         if (creatable && item.create) {
           delete item.create;
+          onCreate?.(value, item, event);
```
Comment (simonguo) [inline src/InputPicker/InputPicker.tsx:317]: You should add corresponding unit tests for the `onCreate` prop.

Comment (simonguo) [reply src/InputPicker/InputPicker.tsx:317]: https://github.com/rsuite/rsuite#test-driven-developmenttdd

Comment (BossBele) [reply src/InputPicker/InputPicker.tsx:317]: On it

Comment (simonguo) [reply src/InputPicker/InputPicker.tsx:317]: https://github.com/rsuite/rsuite/blob/next/src/InputPicker/test/InputPickerSpec.js

Comment (BossBele) [reply src/InputPicker/InputPicker.tsx:317]: @simonguo Added tests. Having problems with `npm install` while testing. Should I submit a bug?

File: src/SelectPicker/SelectPicker.tsx [hunk 2]
```diff
@@ -83,6 +83,9 @@ export interface SelectProps<T = ValueType> {
   /** Called when the option is selected */
   onSelect?: (value: any, item: ItemDataType, event: React.SyntheticEvent) => void;
 
+  /** Called when the option is created */
+  onCreate?: (value: any, item: ItemDataType, event: React.SyntheticEvent) => void;
```
Comment (simonguo) [inline src/SelectPicker/SelectPicker.tsx:87]: `onCreate` is a property of InputPicker, so it should not be declared in SelectPicker.

File: src/InputPicker/test/InputPickerSpec.js [hunk 3]
```diff
@@ -330,4 +330,18 @@ describe('InputPicker', () => {
...
+      if (!(data.includes(value) || data.includes(item.value))) {
+        done();
+      }
+    };
+    const instance = getInstance(
+      <InputPicker defaultOpen data={data} onCreate={doneOp} defaultValue={'Kariane'} creatable />
+    );
+
+    ReactTestUtils.Simulate.keyDown(instance.target, { keyCode: 40 });
+    ReactTestUtils.Simulate.keyDown(instance.target, { keyCode: 13 });
```
Comment (simonguo) [inline src/InputPicker/test/InputPickerSpec.js:345]: Will not trigger `onCreate` without generating new options. Let me solve this problem

File: src/InputPicker/InputPicker.tsx (hunk 1 above)
Comment (simonguo) [reply src/InputPicker/InputPicker.tsx:317]: There is an error in the tests, let me check.

Comment (simonguo) [reply src/InputPicker/InputPicker.tsx:317]: In addition, the documentation update instructions for onCreate are missing.

**Pull request merged**